google/cloud/bigquery/v2/minimal/benchmarks/benchmarks_config.h

```diff
@@ -0,0 +1,102 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGQUERY_V2_MINIMAL_BENCHMARKS_BENCHMARKS_CONFIG_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGQUERY_V2_MINIMAL_BENCHMARKS_BENCHMARKS_CONFIG_H
+
+#include "google/cloud/bigquery/v2/minimal/internal/job_request.h"
+#include "google/cloud/bigquery/v2/minimal/internal/table_view.h"
+#include "google/cloud/status_or.h"
+#include "google/cloud/version.h"
+#include <chrono>
+#include <string>
+#include <vector>
+
+namespace google {
+namespace cloud {
+namespace bigquery_v2_minimal_benchmarks {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+struct Config {
+  std::string base_endpoint;
+  std::string relative_endpoint;
+  std::string endpoint;
+
+  std::string project_id;
+
+  int samples = 2;
+  std::chrono::seconds iteration_duration = std::chrono::seconds(5);
+
+  int minimum_threads = 1;
+  int maximum_threads = 1;
+  int minimum_channels = 1;
+  int maximum_channels = 1;
+
+  std::string page_token;
+  int max_results = 1000;
+
+  google::cloud::StatusOr<Config> ParseArgs(
+      std::vector<std::string> const& args);
+
+ protected:
+  struct Flag {
+    std::string flag_name;
+    std::function<void(std::string)> parser;
+  };
+
+  void ParseCommonFlags();
+  inline bool FlagsParsed() { return !flags_.empty(); }
+  google::cloud::Status ValidateArgs(std::vector<std::string> const& args);
+
+  std::vector<Config::Flag> flags_;
+};
+
+struct DatasetConfig : public Config {
+  std::string filter;
+  bool all;
+
+  google::cloud::StatusOr<DatasetConfig> ParseArgs(
+      std::vector<std::string> const& args);
+};
```

Comment:
To complete this review it would be useful to understand what do you want to benchmark.

Do we really care how long it takes to perform CRUDL operations on datasets?  Would any customer care?  Likewise for tables and even jobs?

I would think that only job results are interesting?  And even then, do we care about end-to-end throughput / latency, including the time spent in the service?  Or do we care about client-side metrics, like CPU time?